### PR TITLE
bloodhound: init at 4.1.0

### PR DESCRIPTION
**Pull request by Lyrain**
###### Description of changes

Adds bloodhound - a utility for visualizing Active Directory to identify security issues within a domain. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Please rebase

**Comment by AkechiShiro** [comment]
Hi, could I take over on this PR ? @Lyrain @onny ?

File: pkgs/applications/misc/bloodhound/default.nix [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  nativeBuildInputs = [ makeWrapper unzip ];
+
+  dontUnpack = true;
+  dontBuild = true;
+  dontPatchELF = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    unzip $src
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bloodhound/default.nix:102]: This should be done in unpackPhase

File: pkgs/applications/misc/bloodhound/default.nix [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  ] + ":${stdenv.cc.cc.lib}/lib64";
+
+  buildInputs = [
+    gtk3 # needed for GSETTINGS_SCHEMAS_PATH
+  ];
+
+  nativeBuildInputs = [ makeWrapper unzip ];
+
+  dontUnpack = true;
+  dontBuild = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bloodhound/default.nix:96]: This should be a noop, so please keep it enabled.

File: pkgs/applications/misc/bloodhound/default.nix [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+, nss
+, pango
+, systemd
+, unzip
+, xdg-utils
+, xorg
+}:
+stdenv.mkDerivation rec {
+  pname = "bloodhound";
+  binaryName = "BloodHound";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bloodhound/default.nix:39]: This should be in meta.mainProgram

File: pkgs/applications/misc/bloodhound/default.nix [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    unzip $src
+    mkdir -p $out
+    mkdir -p $out/{bin,lib/${binaryName}}
+    mv BloodHound-linux-x64/* $out/lib/${binaryName}
+    chmod +x $out/lib/${binaryName}/${binaryName}
+
+    patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
+       $out/lib/${binaryName}/${binaryName}
+
+    makeWrapper $out/lib/${binaryName}/${binaryName} $out/bin/${binaryName} \
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bloodhound/default.nix:111]: Please just hardcode binaryName to reduce complexity

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -7441,6 +7441,12 @@
     githubId = 39707188;
     name = "Lynn Dong";
   };
+  lyrain = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7444]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Comment by AkechiShiro** [comment]
Sorry @SuperSandro2000 I mentioned this PR in ready for review however I plan on either finishing #242534 or packaging the whole building process as well.

**Comment by abathur** [comment]
Closing since bloodhound landed in nixpkgs a month ago via the merge of #242534.

@Lyrain Thanks for the effort you put in to getting this started! If you're still using bloodhound, you may want to open a separate PR to add yourself as maintainer?

**Comment by AkechiShiro** [comment]
Thanks for closing this @abathur I completely forgot to ask for this after having had the merge of BloodHound be done.
